File: ArraysAndStrings/1.2 - Permutation/Permutation.py
def is_permutation(s1, s2):
    s1_srt = sorted(s1)
    s2_srt = sorted(s2)
    perm = True
    i = 0
    while i < len(s2):
        if s1_srt[i] != s2_srt[i]:
            perm = False
        i += 1

    if perm == False:
        print ("Not permutation")
    else:
        print("Permutation")

def main():
    s1 = "abc"
    s2 = "cba"
    s3 = "aab"

    is_permutation(s1,s2)
    is_permutation(s1,s3)

if __name__ == '__main__':
    main()

File: ArraysAndStrings/1.2 - Permutation/Permutation.java
import java.util.Arrays;

public class Permutation {
    static String sortString(String s1){
        char tempArray[] = s1.toCharArray();
        Arrays.sort(tempArray);
        return new String(tempArray);
    }
    static void isPermutation(String s1, String s2){
        boolean perm = True;
        s1 = sortString(s1);
        s2 = sortString(s2);
    }
    public static void main(String args[]){
        String s1 = "abc";
        String s2 = "cba";
        String s3 = "baa";
        isPermutation(s1,s2);
        isPermutation(s1,s3);
    }
}